src/communication/communication_channel.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+/*
+ * communication_channel.cpp - wrapper for communication primitives
+ *                           - it can establish a connection, send/receive messages, wait for events and close the connection
+ */
+
+#include "communication_channel.hpp"
+
+#include "connection_support.h"
+#include "connection_globals.h"
+
+#include "string_buffer.hpp"
+#include "system_parameter.h"
+#if defined(WINDOWS)
+#include "wintcp.h"
```

Comment:
Without dump data, string_buffer & system_parameter are not required.
```suggestion
#include "connection_support.h"
#include "connection_globals.h"

#if defined(WINDOWS)
#include "wintcp.h"
```

Reply:
done